src/sidebar/components/group-list-v2.js

```diff
@@ -0,0 +1,119 @@
+'use strict';
+
+const { createElement } = require('preact');
+const { useMemo } = require('preact/hooks');
+const propTypes = require('prop-types');
+
+const isThirdPartyService = require('../util/is-third-party-service');
+const { isThirdPartyUser } = require('../util/account-id');
+const groupsByOrganization = require('../util/group-organizations');
+const useStore = require('../store/use-store');
+const { withServices } = require('../util/service-context');
+const serviceConfig = require('../service-config');
+
+const Menu = require('./menu');
+const MenuItem = require('./menu-item');
+const GroupListSection = require('./group-list-section');
+
+/**
+ * Return the custom icon for the top bar configured by the publisher in
+ * the Hypothesis client configuration.
+ */
+function publisherProvidedIcon(settings) {
+  const svc = serviceConfig(settings);
+  return svc && svc.icon ? svc.icon : null;
+}
+
+/**
+ * Menu allowing the user to select which group to show and also access
+ * additional actions related to groups.
+ */
+function GroupList({ serviceUrl, settings }) {
+  const currentGroups = useStore(store => store.getCurrentlyViewingGroups());
+  const featuredGroups = useStore(store => store.getFeaturedGroups());
+  const myGroups = useStore(store => store.getMyGroups());
+  const focusedGroup = useStore(store => store.focusedGroup());
+  const userid = useStore(store => store.profile().userid);
+
+  const myGroupsSorted = useMemo(() => groupsByOrganization(myGroups), [
+    myGroups,
+  ]);
+
+  const featuredGroupsSorted = useMemo(
+    () => groupsByOrganization(featuredGroups),
+    [featuredGroups]
+  );
+
+  const currentGroupsSorted = useMemo(
+    () => groupsByOrganization(currentGroups),
+    [currentGroups]
+  );
+
+  const { authDomain } = settings;
+  const canCreateNewGroup = userid && !isThirdPartyUser(userid, authDomain);
+  const newGroupLink = serviceUrl('groups.new');
+
+  let label;
+  if (focusedGroup) {
+    const icon = focusedGroup.organization.logo;
+    label = (
```

Comment:
I wonder if there is any way to tap into the `SVGIcon` component here in the future. I know that right now `SVGIcon` depends on a hard-coded mapping of `name => svg source` but I wonder if it could be expanded upon to make rendering these plausible...

Reply:
`<SvgIcon>` currently renders an `<svg>` inline for CSS style-ability. Here we use an `<img>` a) because this needs to be loaded from a remote source and b) because this is untrusted publisher-provided markup.